### PR TITLE
convert grad_merge_all_reduce in graph to program

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Function optimization
### PR changes

Others
### Describe

convert grad_merge_all_reduce in graph to program

### CODE REVIEW

**Review by LiYuRio — APPROVED** [review]
LGTM for c_all_reduce

**Review by chenwhql — APPROVED** [review]

**Review by Shixiaowei02 — APPROVED** [review]
this modification does not affect inference

File: paddle/fluid/framework/ir/coalesce_grad_tensor_pass.cc [hunk 1]
```diff
@@ -559,38 +559,26 @@ class CoalesceGradTensorPass : public ir::Pass {
...
-      auto *global_block = program_desc.MutableBlock(0);
-      AppendAllocSpaceForVarsOp(params_name,
-                                grads_name,
-                                fused_var_name,
-                                dtype,
-                                any_persistable,
-                                global_block);
-    }
+    VLOG(4) << "all_persistable:" << all_persistable;
+    VLOG(4) << "any_persistable:" << all_persistable;
```
Comment (From00) [inline paddle/fluid/framework/ir/coalesce_grad_tensor_pass.cc:563]: Modify in next PR: all_persistable -> any_persistable

**Review by From00 — APPROVED** [review]

File: paddle/fluid/framework/ir/coalesce_grad_tensor_pass.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/framework/ir/coalesce_grad_tensor_pass.cc:563]: ok

**Pull request merged**